pandas/tests/io/test_excel.py

```diff
@@ -1414,133 +1321,117 @@ def test_excel_date_datetime_format(self):
-
-    def test_to_excel_interval_labels(self):
+        frame = DataFrame(np.random.randint(-10, 10, size=(20, 1)),
+                          dtype=np.int64)
+        expected = frame.copy()
+        frame['new'] = pd.cut(frame[0], 10)
+        expected['new'] = pd.cut(expected[0], 10).astype(str)
+        frame.to_excel(self.path, 'test1')
+        reader = ExcelFile(self.path)
+        recons = read_excel(reader, 'test1')
+        tm.assert_frame_equal(expected, recons)
+
+    def test_to_excel_interval_labels(self, merge_cells, engine, ext):
         # GH19242 - test writing Interval with labels
-        _skip_if_no_xlrd()
-
-        with ensure_clean(self.ext) as path:
-            frame = DataFrame(np.random.randint(-10, 10, size=(20, 1)),
-                              dtype=np.int64)
-            expected = frame.copy()
-            intervals = pd.cut(frame[0], 10, labels=['A', 'B', 'C', 'D', 'E',
-                                                     'F', 'G', 'H', 'I', 'J'])
-            frame['new'] = intervals
-            expected['new'] = pd.Series(list(intervals))
-            frame.to_excel(path, 'test1')
-            reader = ExcelFile(path)
-            recons = read_excel(reader, 'test1')
-            tm.assert_frame_equal(expected, recons)
-
-    def test_to_excel_timedelta(self):
+        frame = DataFrame(np.random.randint(-10, 10, size=(20, 1)),
+                          dtype=np.int64)
+        expected = frame.copy()
+        intervals = pd.cut(frame[0], 10, labels=['A', 'B', 'C', 'D', 'E',
+                                                 'F', 'G', 'H', 'I', 'J'])
+        frame['new'] = intervals
+        expected['new'] = pd.Series(list(intervals))
+        frame.to_excel(self.path, 'test1')
+        reader = ExcelFile(self.path)
+        recons = read_excel(reader, 'test1')
+        tm.assert_frame_equal(expected, recons)
+
+    def test_to_excel_timedelta(self, merge_cells, engine, ext):
         # GH 19242, GH9155 - test writing timedelta to xls
-        _skip_if_no_xlrd()
-
-        with ensure_clean('.xls') as path:
-            frame = DataFrame(np.random.randint(-10, 10, size=(20, 1)),
-                              columns=['A'],
-                              dtype=np.int64
-                              )
-            expected = frame.copy()
-            frame['new'] = frame['A'].apply(lambda x: timedelta(seconds=x))
-            expected['new'] = expected['A'].apply(
-                lambda x: timedelta(seconds=x).total_seconds() / float(86400))
-            frame.to_excel(path, 'test1')
-            reader = ExcelFile(path)
-            recons = read_excel(reader, 'test1')
-            tm.assert_frame_equal(expected, recons)
-
-    def test_to_excel_periodindex(self):
-        _skip_if_no_xlrd()
-
+        if engine == 'openpyxl':
+            pytest.xfail('Timedelta roundtrip broken with openpyxl')
+        if engine == 'xlsxwriter' and (sys.version_info[0] == 2 and
```

Comment:
This keeps failing on Travis with Py2, but AppVeyor was fine and I can't reproduce locally (macOS) so I assume it to be linux-only (?). Since this is a "new" test I figure just skip for now and I'll open an issue to see if someone on a Linux platform can help troubleshoot. If you have a better idea on how to manage let me know